test/utils/fixture/fixture.go

```diff
@@ -94,13 +96,63 @@ func SetupWithOptions(name string, nrtList nrtv1alpha2.NodeResourceTopologyList,
 	}
 	randomizeName := (options & OptionRandomizeName) == OptionRandomizeName
 	avoidCooldown := (options & OptionAvoidCooldown) == OptionAvoidCooldown
+	isSuiteSetup := (options & OptionIsSuite) == OptionIsSuite
+	ginkgo.By("set up the test namespace")
 	ns, err := setupNamespace(e2eclient.Client, name, randomizeName)
 	if err != nil {
 		klog.Errorf("cannot setup namespace %q: %v", name, err)
 		return nil, err
 	}
-	ginkgo.By(fmt.Sprintf("set up the test namespace %q", ns.Name))
+	klog.Infof("test namespace %q was set up successfully", ns.Name)
+	klog.Info("Going Once .... \n\n\n")
+	if !isSuiteSetup {
+		ginkgo.By("pull NRT data before test starts")
+		var nrtAtTestSetup nrtv1alpha2.NodeResourceTopologyList
+		immediate := true
+		err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 30*time.Second, immediate, func(ctx context.Context) (bool, error) {
+			err := e2eclient.Client.List(context.TODO(), &nrtAtTestSetup)
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			klog.Errorf("failed to pull NRT items: %v", err)
+			return nil, err
+		}
+
+		ginkgo.By("verify the collected NRT data is similar to the data gathered in the beginning of the suite")
+		nrtMismatch := false
+		for _, nrtCurrent := range nrtAtTestSetup.Items {
+			nrtBefore, err := e2enrt.FindFromList(nrtList.Items, nrtCurrent.Name)
+			if err != nil {
+				klog.Errorf("failed to find NRT item %q in NRT list gathered at the suite setup: %v", nrtCurrent.Name, err)
+				nrtMismatch = true
+				break
+			}
+			ok, err := e2enrt.CheckEqualAvailableResources(*nrtBefore, nrtCurrent)
+			if err != nil {
+				klog.Warningf("failed to compare NRT items: %v", err)
+				continue
+			}
+			if !ok {
+				klog.Warningf("WARNING! NRT data for %q at the beginning of this test is different from the NRT data gathered at the beginning of the suite", nrtCurrent.Name)
+				nrtMismatch = true
+			}
+			klog.Info("Going loop .... \n")
+
+		}
+		if nrtMismatch {
+			klog.Warning("NRT MISMATCH:\n")
+			a, _ := yaml.Marshal(nrtList.Items)
+			klog.Infof("-----NRT at Suite Setup: \n%s\n", a)
+			b, _ := yaml.Marshal(nrtAtTestSetup.Items)
+			klog.Infof("-----NRT at Test Setup: \n%s\n", b)
+		}
```

Comment:
this should probably be put in a separate function. Also, please check functions in `internal/noderesourcetopology/tostring.go`